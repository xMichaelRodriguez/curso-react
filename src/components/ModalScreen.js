import React, { useState } from "react";
import LoadingMask from "react-loadingmask";
import "react-loadingmask/dist/react-loadingmask.css";
import Modal from "react-modal";
import { useDispatch, useSelector } from "react-redux";
import { startDeletedImage } from "../actions/images";
import { uiCloseModal } from "../actions/uiActions";

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    padding: "0",
    transform: "translate(-50%, -50%)",
  },
};

// Make sure to bind modal to your appElement (https://reactcommunity.org/react-modal/accessibility/)
Modal.setAppElement("#root");

export const ModalSchreen = () => {
  const dispatch = useDispatch();
  const [Loading, setLoading] = useState(false);
  const [deleted, setDeleted] = useState("");
  const { modalOpen } = useSelector((state) => state);
  let subtitle;

  const afterOpenModal = () => {
    // references are now sync'd and can be accessed.
    subtitle.style.color = "#fffff";
  };

  const closeModal = () => {
    dispatch(uiCloseModal());
  };

  const handleDelete = async () => {
    setLoading(true);
    const resp = await dispatch(startDeletedImage());
    if (resp.ok) {
      setDeleted(resp.msg);
      closeModal();
    }

    setLoading(false);
  };
  return (
    <Modal
      isOpen={modalOpen}
      onAfterOpen={afterOpenModal}
      onRequestClose={closeModal}
      style={customStyles}
      contentLabel="Delete Modal"
      overlayClassName="modal-fondo"
    >
      <div className="card">
        <div className="card-header">
          <button
            onClick={closeModal}
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
          <h2
            className="card-title"
            ref={(_subtitle) => (subtitle = _subtitle)}
          >
            Eliminacion de Imagen
          </h2>
        </div>
        <div className="card-body">
          {!!deleted ? (
            <h3>{deleted}</h3>
          ) : Loading ? (
            <div>
              <LoadingMask
                style={{ backgroundColor: "#999999", heigth: "100%" }}
                loading={Loading}
                text={"loading..."}
              >
                <h3>Estas seguro de eliminarla?</h3>
              </LoadingMask>
            </div>
          ) : (
            <h5>Estas seguro de eliminarla?</h5>
          )}
        </div>
        <div className="card-footer d-flex justify-content-between">
          <button
            type="button"
            className="btn btn-primary"
            onClick={handleDelete}
          >
            Si, borrala
          </button>
          <button
            type="button"
            className="btn btn-secondary"
            onClick={closeModal}
          >
            cerrar
          </button>
        </div>
      </div>
    </Modal>
  );
};
