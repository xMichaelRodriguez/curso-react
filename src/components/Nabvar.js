import React from "react";

export const Nabvar = () => {
  return (
    <div>
      <nav className="navbar navbar-light bg-dark">
        <div className="container-fluid d-flex justify-content-center">
          <div className="input-group ">
            <form className="form-inline my-2 my-md-0 w-100 d-flex justify-content-center">
              <input
                className="form-control mr-sm-2 w-75"
                type="text"
                placeholder="Search"
              />
              <button className="btn btn-secondary  rounded-circle my-2 my-sm-0" type="button" >
              <i className="fas fa-search"></i>
              </button>
            </form>
          </div>
        </div>
      </nav>
    </div>
  );
};
