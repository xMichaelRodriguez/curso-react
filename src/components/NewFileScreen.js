import React, { useState } from "react";

import LoadingMask from "react-loadingmask";
import "react-loadingmask/dist/react-loadingmask.css";

import { fetchAsync } from "../helpers/querys";

export const NewFileScreen = () => {
  const [FormValue, setFormValue] = useState("");
  const [Loading, setLoading] = useState(false);

  const handleInputChange = (e) => {
    e.preventDefault();

    setFormValue(e.target.files[0]);
  };

  const handleUpload = async () => {
    if (FormValue.length <= 0) {
      alert("Selecciona una imagen");
    } else {
      setLoading(true);

      const resp = await fetchAsync("newFile", FormValue, "POST");
      const body = await resp.json();
      console.log(body);
      setLoading(false);
      alert("imagen subida", body);
    }
  };
  return (
    <div>
      {Loading ? (
        <LoadingMask
          style={{ backgroundColor: "#ffffff" }}
          loading={Loading}
          text={"loading..."}
        >
          <div className="card bg-dark" style={{ height: 300 }}>
            <div className="card-header text-white    ">
              <h1>Uploading Image</h1>
            </div>
            <div className="card-body">
              <div className="mb-3">
                <div className="mb-3">
                  <input
                    className="form-control"
                    type="file"
                    name="image"
                    onChange={handleInputChange}
                  />
                </div>
              </div>
              <div
                className="btn btn-primary d-flex inline justify-content-center"
                onClick={handleUpload}
              >
                <box-icon
                  name="cloud-upload"
                  type="solid"
                  color="#fff"
                ></box-icon>
                <span>Upload</span>
              </div>
            </div>
          </div>
        </LoadingMask>
      ) : (
        <div className="card bg-dark" style={{ height: 300 }}>
          <div className="card-header text-white    ">
            <h1>Uploading Image</h1>
          </div>
          <form
            id="form"
            className="card-body"
            onSubmit={handleUpload}
            encType="multipart/form-data"
          >
            <div className="mb-3">
              <div className="mb-3">
                <input
                  className="form-control"
                  type="file"
                  name="image"
                  accept="image/jpg,.jpeg,.png"
                  onChange={handleInputChange}
                />
              </div>
            </div>
            <div className="d-grid gap-2 d-md-block">
              <button type="submit" className="btn btn-primary fs-2">
                <box-icon
                  name="cloud-upload"
                  type="solid"
                  color="#fff"
                  size="sm"
                  pull="left"
                ></box-icon>
                <span> Upload</span>
              </button>
            </div>
          </form>
        </div>
      )}
    </div>
  );
};
