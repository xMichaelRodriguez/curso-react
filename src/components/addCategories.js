import React, { useState } from "react";

const AddCategories = () => {
  const [inputValue, setInputValue] = useState("Hola Mundo");
  const inputChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    console.log(inputValue)
  };
  return (
    <form onSubmit={handleSubmit}>
      <h2>{inputValue}</h2>
      <input type="text" value={inputValue} onChange={inputChange} />
    </form>
  );
};

export default AddCategories;
