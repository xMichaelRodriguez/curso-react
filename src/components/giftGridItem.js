import React from "react";
import PropTypes from "prop-types"; // ES6
export const GiftGridItem = ({ title, url }) => {
  return (
    <>
      <div className="card-image animate__animated animate__flipInX">
        <img src={url} alt={title} height="150px" width="150px" />
        <span className="card-title">{title}</span>
      </div>
    </>
  );
};

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
