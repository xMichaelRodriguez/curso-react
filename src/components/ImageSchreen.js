import React from "react";
import { useDispatch } from "react-redux";
import { uiOpenModal } from "../actions/uiActions";
import { ModalSchreen } from "./ModalScreen";
export const ImageSchreen = ({ image, id }) => {
  const dispatch = useDispatch();

  const handleImageClick = () => {
    dispatch(uiOpenModal());
  };
  return (
    <>
      <div className="card shadow text-white">
        <img
          className="card-img-top"
          onDoubleClick={handleImageClick}
          src={image}
          key={id}
          alt="...."
        ></img>
      </div>

      <ModalSchreen item={id} />
    </>
  );
};
