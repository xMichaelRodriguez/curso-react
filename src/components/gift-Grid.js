import React from "react";
import { GiftGridItem } from "./giftGridItem";
import { useFetchGifts } from "./hooks/useFetchGifts";

export const GiftGrid = ({ categories }) => {
  const { data: images, loading } = useFetchGifts(categories);

  return (
    <>
      <h3>{categories}</h3>

      {loading && <p className=" animate__animated animate__flash">Loading...</p>}
      <div className="container">
        <div className="row ">
          {images.map((img) => (
            <GiftGridItem key={img.id} {...img} />
          ))}
        </div>
      </div>
    </>
  );
};
