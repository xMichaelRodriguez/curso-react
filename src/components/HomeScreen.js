import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { startImageLoading } from "../actions/images";
import { ImageSchreen } from "./ImageSchreen";

export const HomeScreen = () => {
  const dispatch = useDispatch();
  const imagesStore = useSelector((state) => state.images);

  useEffect(() => {
    dispatch(startImageLoading(imagesStore));
  }, [dispatch]);

  return (
    <>
      <div className="row">
        <div className="card-columns">
          {imagesStore !== "" &&
            imagesStore.map(({ image, id }) => (
              <ImageSchreen image={image} id={id} key={id} />
            ))}
        </div>
      </div>

      <Link to="/newFile" className="fabs btn btn-info rounded-circle ">
        <i className="fas fa-plus fa-3x"></i>
      </Link>
    </>
  );
};
