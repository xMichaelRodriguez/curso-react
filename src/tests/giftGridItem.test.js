import React from "react";
import { shallow } from "enzyme";
import { GiftGridItem } from "../components/giftGridItem";

describe("Debe retornar el componente <GiftGridItem />", () => {
  const title = "Hola papi";
  const url = "http://localhost/algo.jpg";

  const wrapper = shallow(<GiftGridItem title={title} url={url} />);

  test("Debe mostrar el componenete correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un parrafo con el TITLE", () => {
    const p = wrapper.find("span").text().trim();

    expect(p).toBe(title);
  });
  test("Debe tener la imagen igual a la url  de los props", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener animate__flipInX", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__flipInX")).toBe(true);
  });
});
