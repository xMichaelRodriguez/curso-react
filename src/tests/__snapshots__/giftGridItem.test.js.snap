// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe retornar el componente <GiftGridItem /> Debe mostrar el componenete correctamente 1`] = `
<Fragment>
  <div
    className="card-image animate__animated animate__flipInX"
  >
    <img
      alt="Hola papi"
      height="150px"
      src="http://localhost/algo.jpg"
      width="150px"
    />
    <span
      className="card-title"
    >
      Hola papi
    </span>
  </div>
</Fragment>
`;
